x-pack/test/apm_api_integration/tests/mobile/mobile_main_statistics_by_field.spec.ts

```diff
@@ -0,0 +1,106 @@
+  const registry = getService('registry');
+  const synthtraceEsClient = getService('synthtraceEsClient');
+
+  const start = new Date('2023-01-01T00:00:00.000Z').getTime();
+  const end = new Date('2023-01-01T00:15:00.000Z').getTime() - 1;
+
+  async function getMobileMainStatisticsByField({
+    environment = ENVIRONMENT_ALL.value,
+    kuery = '',
+    serviceName,
+    field,
+  }: {
+    environment?: string;
+    kuery?: string;
+    serviceName: string;
+    field: string;
+  }) {
+    return await apmApiClient
+      .readUser({
+        endpoint: 'GET /internal/apm/mobile-services/{serviceName}/main_statistics',
+        params: {
+          path: { serviceName },
+          query: {
+            environment,
+            start: new Date(start).toISOString(),
+            end: new Date(end).toISOString(),
+            kuery,
+            field,
+          },
+        },
+      })
+      .then(({ body }) => body);
+  }
+
+  registry.when(
+    'Mobile main statistics when data is not loaded',
+    { config: 'basic', archives: [] },
+    () => {
+      describe('when no data', () => {
+        it('handles empty state', async () => {
+          const response = await getMobileMainStatisticsByField({
+            serviceName: 'foo',
+            field: 'service.version',
+          });
+          expect(response.mainStatistics.length).to.be(0);
+        });
+      });
+    }
+  );
+
+  registry.when('Mobile main statistics', { config: 'basic', archives: [] }, () => {
+    before(async () => {
+      await generateMobileData({
+        synthtraceEsClient,
+        start,
+        end,
+      });
+    });
+
+    after(() => synthtraceEsClient.clean());
+
+    describe('when data is loaded', () => {
+      it('returns the correct data for App version', async () => {
+        const response = await getMobileMainStatisticsByField({
+          serviceName: 'synth-android',
+          environment: 'production',
+          field: 'service.version',
+        });
+        const fieldNames = response.mainStatistics.map((item) => item.name);
+        expect(fieldNames.length).to.be.equal(3);
+      });
+      it('returns the correct data for Os version', async () => {
+        const response = await getMobileMainStatisticsByField({
+          serviceName: 'synth-android',
+          environment: 'production',
+          field: 'host.os.version',
+        });
+        const fieldNames = response.mainStatistics.map((item) => item.name);
+        expect(fieldNames.length).to.be.equal(1);
+      });
+      it('returns the correct data for Devices', async () => {
+        const response = await getMobileMainStatisticsByField({
+          serviceName: 'synth-android',
+          environment: 'production',
+          field: 'device.model.identifier',
+        });
+        const fieldNames = response.mainStatistics.map((item) => item.name);
+        expect(fieldNames.length).to.be.equal(3);
+      });
+    });
```

Comment:
Should this also test correctness of metrics? Looks like it's a very simple count of items returned now, at the very least it should probably test _which_ items are returned by the API?

Reply:
I'll check how to improve these tests

Reply:
So, I'll change it to something like this 
`expect(fieldNames).to.be.eql(['SM-G973F', 'HUAWEI P2-0000', 'SM-G930F']);`
What do you think @dgieselaar ?

Reply:
yes but please also test for the other statistics, they're in the same API response presumably?

Reply:
I added other statistics, waiting to test the launch time and crashes until we know better how to generate the sythtrace data for it